flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/config/ExecutionConfigOptions.java

```diff
@@ -336,15 +337,20 @@
                                     + "\"SortMergeJoin\", \"HashAgg\", \"SortAgg\".\n"
                                     + "By default no operator is disabled.");
 
+    @Deprecated
     @Documentation.TableOption(execMode = Documentation.ExecMode.BATCH)
     public static final ConfigOption<String> TABLE_EXEC_SHUFFLE_MODE =
             key("table.exec.shuffle-mode")
                     .stringType()
-                    .defaultValue("ALL_EDGES_BLOCKING")
+                    .noDefaultValue()
                     .withDescription(
                             Description.builder()
                                     .text("Sets exec shuffle mode.")
                                     .linebreak()
+                                    .text(
+                                            "This option is deprecated. Use the core option '%s' instead.",
```

Comment:
Deprecated options are excluded from the documentation, so no one's gonna read this? (maybe put it in the javadocs instead)